src/canvastools/ts/CanvasTools/Core/RegionData.ts

```diff
@@ -145,6 +145,39 @@ export class RegionData implements IRegionData, IMovable, IResizable {
         return area;
     }
 
+    public getLineSegments(): Array<[Point2D, Point2D]> {
+        const points = this.regionPoints;
+        if (points.length < 2) {
+            return []
+        }
+        if (points.length === 2) {
+            return [[points[0], points[1]]]
+        }
+        const segments = [];
+        const pointsLength = points.length;
+        const loopLength = pointsLength - 1;
+        for (let i = 0; i < loopLength; i++) {
+            const nextPointIdx = i + 1;
+            if (nextPointIdx < pointsLength) {
+                segments.push([points[i], points[nextPointIdx]])
+            }
+        }
+        if (this.regionType == RegionDataType.Polygon) {
+            // closing line segment from last to first point
+            segments.push([points[pointsLength - 1], points[0]])
+        }
+        return segments;
+    }
+
+    public getLineMidpoints(): Point2D[] {
+        const lines = this.getLineSegments();
+        return lines.map(line => {
+            const x = line[0].x - ((line[0].x - line[1].x) / 2);
```

Comment:
line[i].x is always greater than line[i+1].x ?

Reply:
If not, we need to sort. If yes, who maintains order? Is it just inserted in order and our code just always maintains it?

Reply:
Points are added into the array in the order they are added during the polygon creation.
Lines are calculated by order of the points in the array, creating a "DAG" of sorts with a single path of vertices and edges.

This method starts at the beginning of this path and measure out a halfway point between each vertex along the way.